### PR TITLE
Message for no access error is displayed at every other attempt for code completion

### PR DESCRIPTION
**Pull request by manstis**
Ensure the "[feedback] in progress" flag is reset in all scenarios; success and error.

### CODE REVIEW

File: src/features/lightspeed/api.ts [hunk 1]
```diff
@@ -160,7 +160,12 @@ export class LightSpeedAPI {
       }
       return {} as CompletionResponseParams;
     } finally {
-      if (isCompletionSuccess && !this.cancelSuggestionFeedback(suggestionId)) {
```
Comment (TamiTakamiya) [inline src/features/lightspeed/api.ts:163]: I see... Thank you for finding this out!!

**Review by TamiTakamiya — APPROVED** [review]
LGTM. Thanks!

**Comment by hasys** [comment]
Looks like tests do not have 

```
assert.equal(cancelSuggestionFeedbackSpy.called, false);
```

scenario only `true` cases. Is it intended?

**Comment by manstis** [comment]
@hasys 

> Looks like tests do not have
> 
> ```
> assert.equal(cancelSuggestionFeedbackSpy.called, false);
> ```
> 
> scenario only `true` cases. Is it intended?

Yes, see https://github.com/ansible/vscode-ansible/blob/71915d564deaf8420eb8a052f09f34a22dc38905/src/features/lightspeed/api.ts#L163

We *always* ensure the in-flight/in-progress feedback is reset.

We [used to](https://github.com/ansible/vscode-ansible/blob/main/src/features/lightspeed/api.ts#L163) only reset it if the "completion" call was successful; which led to the problem @TamiTakamiya reported.

**Comment by hasys** [comment]
> @hasys
> 
> > Looks like tests do not have
> > ```
> > assert.equal(cancelSuggestionFeedbackSpy.called, false);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > scenario only `true` cases. Is it intended?
> 
> Yes, see
> 
> https://github.com/ansible/vscode-ansible/blob/71915d564deaf8420eb8a052f09f34a22dc38905/src/features/lightspeed/api.ts#L163
> 
> We _always_ ensure the in-flight/in-progress feedback is reset.
> 
> We [used to](https://github.com/ansible/vscode-ansible/blob/main/src/features/lightspeed/api.ts#L163) only reset it if the "completion" call was successful; which led to the problem @TamiTakamiya reported.

If it's always true, should it be removed entirely instead of ensuring it is always true?

**Comment by manstis** [comment]
@hasys 

> If it's always true, should it be removed entirely instead of ensuring it is always true?

The problem @TamiTakamiya found was caused by it *not* being called for both (success and failed) scenarios.

The tests are now asserting that the method was called. Isn't this the point of the test, to check something happened?

**Pull request merged**